### DISCUSSION TITLE
Supporting banking for as many banks as possible

### DISCUSSION THREAD

**Discussion by LefterisJP** [rotki/rotki#4640] [Ideas]
# Intro

This is a disussion to discuss and document various approach to solving the problem of banking tracking in rotki.

# Methods

There is many combinations of technologies to follow in order to support retrieval of banking data. Almost of these methods would need 2FA.

## Pure scraping

This always works, but is a lot of work to maintain per bank as interfaces can change. Will also always need 2fa.

## PSD2

https://www.bundesbank.de/en/tasks/payment-systems/psd2/psd2-775954

This is unfortunately a system only for companies which create middlemen services to talk with banks. And then you can register with those services to talk with your bank. And you need to pay to do so. An example app that does so: https://moneymoney-app.com/psd2/

## Fints

Along with scraping Fints is probably the most well aligned system with the ideals of consumer-facing local apps. It's used by gnucash https://www.gnucash.org/. To implement fints gnucash uses aqbanking https://github.com/aqbanking/aqbanking.

# Related issues

We have started with creating an initial support in rotki by scraping for specific banks: https://github.com/rotki/rotki/issues/1157

**Comment by AlesRFK** [comment]
Sounds good but for what purpose? Something like https://budgetbakers.com?

**Reply by reeserj** [reply]
Plaid isn't open source but it would handle most connections for the US users.

It also has a fee attached.

**Comment by john-frandsen** [comment]
Good breakdown of the options. A few notes from building in this space (disclosure: I maintain open-banking.io):

**Why PSD2 feels heavy**: The blocker isn't the protocol, it's the certificates. To connect directly you register as a TPP and obtain eIDAS-qualified certs — QWAC for TLS transport auth, QSeal for request signing. That's regulatory onboarding per bank (or via an ASPSP hub), months of paperwork, and non-trivial cost before a single transaction lands. For a small OSS project it's a non-starter.

**The practical workaround**: A regulated AISP can front those eIDAS certs and expose a single normalized API — accounts, balances, and transactions as JSON across hundreds of EU banks. Your client never holds banking certificates. This is the model that makes broad coverage feasible without a scraping army.

**The part that matters for rotki**: rotki is privacy-first and local, so the key question is whether the aggregator sees data in cleartext. Most do. A few now do end-to-end encryption with client-held keys, so the server is effectively a blind relay and can't read balances or transactions. That's the only design that doesn't undercut rotki's promise.

**Cost**: roughly €3-10/mo per end user. Cheaper long-term than maintaining per-bank scrapers, and far less brittle.

FinTS stays the cleanest zero-cost path for German banks. But for broad multi-bank coverage without scraping, a cert-free aggregation layer with client-side encryption is the pragmatic middle ground between FinTS and full PSD2 onboarding.

**Reply by LefterisJP** [reply]
I answered in the issue you opened why I don't agree with our assesment on privacy and your approach: https://github.com/rotki/rotki/issues/12567#issuecomment-4929827015
